common/persistence/dataInterfaces.go

```diff
@@ -760,6 +760,15 @@ type (
 		Encoding common.EncodingType // optional binary encoding type
 	}
 
+	// WorkflowEvents is used as generic workflow history events transaction container
+	WorkflowEvents struct {
```

Comment:
How about:

- Naming this WorkflowEventBatch
- Then creating another type called WorkflowEvents { []WorkflowEventBatch} 

Each of the two types can have a Size() method which returns the size corresponding to the type. That way, you can avoid returning 3 parameters from methods like historyBuilder.ToSerializedEvents()